automation/recipes/factory.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+import os
+import subprocess
+import sys
+from typing import Iterable, Tuple
+import uuid
+
+from utils import SatelliteConfiguration
+
+if sys.version_info[0] == 3 and sys.version_info[1] >= 8:
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+RECIPE_DIR = os.path.split(os.path.abspath(__file__))[0]
+BASE_DIR = "/tmp/observations_tmp"
+os.makedirs(BASE_DIR, exist_ok=True)
+
+def get_recipe(sat: SatelliteConfiguration) -> str:
+    if "recipe" in sat:
+        recipe = sat["recipe"]
+    elif sat["name"].startswith("NOAA"):
+        recipe = "noaa-apt"
+    else:
+        raise LookupError("Unknown recipe")
+
+    recipe_filename = os.path.join(RECIPE_DIR, recipe + ".sh")
+    if os.path.exists(recipe_filename):
+        return recipe_filename
+    else:
+        raise LookupError("Recipe not exists")
+
+def execute_recipe(sat: SatelliteConfiguration, los: datetime.datetime) -> Iterable[Tuple[Literal["signal", "product"], str]]:
```

Comment:
Comment is needed for this function as well. Something like "executes specified recipe. Continues listening until los is reached. The results are then submitted (as specified in the config file)."